WordPress/Classes/Models/BasePost.m

```diff
@@ -181,7 +182,7 @@ - (void)setDateCreated:(NSDate *)localDate
      If the date is now or in the past, and the status is scheduled, set the status
      to published.
      */
-    if (self.date_created_gmt == nil) {
+    if (self.date_created_gmt == nil || [self.date_created_gmt isEqualToDate:self.dateModified]) {
```

Comment:
The conditions inside this block are almost the same as the conditions checked [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/5483/files#diff-d03640d7facc9ecd8eb48be69bf8a577R293).

If feels like these two conditions should be a new method in `BasePost` instead, in order to ease code maintenance.